Untanky/message-queueing
# Sync data and requests from main to replicas

Syncs and replicates data from the main node to multiple replica node. 

To test:

1. start two terminals: 
   - `go run ./cmd -hostname localhost -main -dataDir data/node1` (main node)
   - `go run ./cmd -port 4039 -hostname localhost:4039 -dataDir data/node2` (replica node)
2. send `api/v1` requests towards `localhost:8080` (main node)
3. send `internal` requests towards both hosts (`localhost:8080` and `localhost:4039`) to see that the data is synced

Fixes #2
Fixes #3
Fixes #4
